go/interpreter/lfvm/memory_test.go

```diff
@@ -185,3 +185,120 @@ func TestMemory_expandMemory_expandsMemoryOnlyWhenNeeded(t *testing.T) {
 		})
 	}
 }
+
+func TestMemory_SetByte_SuccessfulCases(t *testing.T) {
+
+	baseData := []byte{0x12, 0x34, 0x56, 0x78, 0x90, 0xab, 0xcd, 0xef}
+
+	tests := map[string]struct {
+		memory []byte
+		offset uint64
+	}{
+		"write to empty memory": {},
+		"write to first element of memory": {
+			memory: baseData,
+			offset: 0,
+		},
+		"write with offset within memory": {
+			memory: baseData,
+			offset: 4,
+		},
+		"offset trigger expansion of empty memory": {
+			offset: 2,
+		},
+		"offset trigger memory expansion": {
+			memory: baseData[:4],
+			offset: 8,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctxt := getEmptyContext()
+			m := NewMemory()
+			m.store = test.memory
+			ctxt.gas = tosca.Gas(3)
+			value := byte(0x12)
```

Comment:
nit: In the first test, byte [0] in the memory has already the value "0x12" before the update.